GeometryService/src/VirtualDetectorMaker.cc

```diff
@@ -635,9 +653,10 @@ namespace mu2e {
       }
 
       if ( c.getBool("vd.STMDet1UpStr.build", false) ) { // VD89, just upstream of LaBr
+        STM_SSC const & pSTM_SSCParams = *stmgh.getSTM_SSCPtr();
         LaBrDetector const & pLaBrDetectorParams = *stmgh.getLaBrDetectorPtr();
         double VD89_offset = 1.0;
-        double xOffset = pLaBrDetectorParams.originInMu2e().x()-stmDnStrEnvPositionInMu2e.x(); // Add collimator hole and offset shifts
+        double xOffset = -3904 + pSTM_SSCParams.offset_Spot() - stmDnStrEnvPositionInMu2e.x();
```

Comment:
It would be better to avoid hard-coding this number (-3904) and use "_config.getDouble("mu2e.solenoidOffset");" from the geometry files to access this value.